src/lib/peer.js

```diff
@@ -79,9 +91,11 @@ function Peer(config) {
 
     function router(envelope) {
         var nextHop = self.fingerTable.bestCandidate(envelope.dstId);
-        log('nextHop:', nextHop, envelope);
-        if (nextHop === self.peerId.toHex()) {
+        console.log('nextHop:', nextHop, envelope);
+        if (nextHop === self.peerId.toHex() && envelope.dstId === self.peerId.toHex()) {
```

Comment:
If the dstId is our peer, fingerTable will return our Id , so when we do `nextHop === self.peerId.toHex()`, it will be the same as `envelope.dstId === self.peerId.toHex()`